app/src/main/java/me/saket/dank/widgets/IndentedLayout.java

```diff
@@ -54,31 +94,57 @@ public IndentedLayout(Context context, @Nullable AttributeSet attrs) {
   protected void onLayout(boolean changed, int l, int t, int r, int b) {
     super.onLayout(changed, l, t, r, b);
 
-    indentationLinePath.reset();
     for (int i = 0; i < indentationDepth; i++) {
       float lineStartX = spacePerDepthPx * (i + 1) + indentationLinePaint.getStrokeWidth();
-      indentationLinePath.moveTo(lineStartX, 0);
-      indentationLinePath.lineTo(lineStartX, getHeight());
+
+      this.trees.get(i).path.moveTo(lineStartX, 0);
+      this.trees.get(i).path.lineTo(lineStartX, getHeight());
     }
   }
 
   @Override
   public void draw(Canvas canvas) {
     super.draw(canvas);
-    canvas.drawPath(indentationLinePath, indentationLinePaint);
+
+    for (ColoredTree tree : trees) {
+      indentationLinePaint.setColor(tree.color);
+      canvas.drawPath(tree.path, indentationLinePaint);
+    }
   }
 
   public void setIndentationDepth(@IntRange(from = 0, to = 1) int depth) {
     indentationDepth = depth;
 
+    this.setupDepthLines(depth);
+
     int indentationSpacing = (int) (indentationDepth * spacePerDepthPx + indentationLinePaint.getStrokeWidth());
     setPaddingRelative(originalPaddingStart + indentationSpacing, getPaddingTop(), getPaddingEnd(), getPaddingBottom());
 
     invalidate();
   }
 
+  private void setupDepthLines(int depth) {
+    trees = new ArrayList<>();
```

Comment:
This one I'm not sure about so mind that but instead of creating new array every time we could just use `clear` method to remove old data. I'll definitely read about that later to confirm whether I'm right or wrong, perhaps it's better to create new array instead like you do now so no change might be needed.

Reply:
I think there are not too many objects to notice a difference between .new() and .clear() but I'm not sure.